superset-frontend/src/views/CRUD/alert/AlertReportModal.tsx

```diff
@@ -554,6 +554,7 @@ const AlertReportModal: FunctionComponent<AlertReportModalProps> = ({
 
     const data: any = {
       ...currentAlert,
+      type: isReport ? 'Report' : 'Alert',
```

Comment:
ensures correct type is always sent to api. There are various code paths updating currentAlert, It's hard to tell which is causing this value to go out of sync with the UI.